libc-top-half/musl/src/env/clearenv.c

```diff
@@ -7,6 +7,10 @@ weak_alias(dummy, __env_rm_add);
 
 int clearenv()
 {
+#ifdef __wasilibc_unmodified_upstream // Lazy environment variable init.
+#else
+#include "wasi/libc-environ-compat.h"
```

Comment:
I'd add a comment here (& in the other uses) indicating that this header is intended to be used inside a func body, and it redefines the __environ symbol. The latter seems like it may cause unintended consequences if someone performed otherwise innocent-looking code motion without understanding the details of that header.

Reply:
Comments added. The most common cases of code motion are guarded against, because if you use `__environ` before it's defined, you'll get errors, and if you use the redefined `__environ` in a function other than the one the header was included in, `__wasilibc_environ` won't be declared and you'll get errors. But I agree, this is particularly tricky code, so comments are good.